build.gradle

```
@@ -3371,7 +3369,7 @@ project(':jmh-benchmarks') {
       if (System.getProperty("jmhArgs")) {
           args System.getProperty("jmhArgs").split(' ')
       }
-      args = [shadowJar.archivePath, *args]
+      args = [shadowJar.archiveFile.get().asFile, *args]
```

For my knowledge should it be as below: https://docs.gradle.org/current/kotlin-dsl/gradle/org.gradle.api.file/-regular-file/index.html?

```suggestion
      args = [shadowJar.archiveFile.get().getAsFile, *args]
```

---

I think both approch could work, please take a look at `layout.buildDirectory.get().asFile.path`, the method also defined `getAsFile` but we can use the `asFile`.

```
public interface Directory extends FileSystemLocation {
    /**
     * Returns the location of this directory, as an absolute {@link File}.
     *
     * @since 4.2
     */
    @Override
    File getAsFile();
 ```